README.rst

```diff
@@ -0,0 +1,760 @@
+differences in syntax, naming and conventions:
+
+- Grafeas uses a scheme, namespace, name and version in a URL-like string
+- Here.com OSRK uses a package manager, name and version field and a colon-
+  separated URL-like string
+- JFrog XRay uses a scheme, namespace, name and version in a URL-like string
+- Libraries.io uses a platform, name and version
+- OpenShift fabric8 analytics uses ecosystem, name and version
+- ScanCode and AboutCode.org use a type, name and version
+- SPDX has an appendix for external repository references and uses a type and a
+  locator with a type-specific syntax for component separators in a URL-like
+  string
+- versioneye uses a type, name and version
+
+
+Solution
+========
+
+A `purl` or package URL is an attempt to standardize existing approaches to
+reliably identify and locate software packages.
+
+A `purl` is a URL string used to identify and locate a software package in a
+mostly universal and uniform way across programing languages, package managers,
+packaging conventions, tools, APIs and databases.
+
+Such a package URL is useful to reliably reference the same software package
+using a simple and expressive syntax and conventions based on familiar URLs.
+
+
+purl
+~~~~~
+
+`purl` stands for **package URL**.
+
+A `purl` is a URL composed of six components::
+
+    type:namespace/name@version?qualifiers#subpath
+
+Components are separated by a specific character for unambiguous parsing.
+
+The defintion for each components is:
+
+- **type**: the package "type" or package "protocol" such as maven, npm, nuget,
+  gem, pypi, etc. Required.
+- **namespace**: some name prefix such as a Maven groupid, a Docker image owner,
+  a GitHub user or organization. Optional and type-specific.
+- **name**: the name of the package. Required.
+- **version**: the version of the package. Optional.
+- **qualifiers**: extra qualifying data for a package such as an OS,
+  architecture, a distro, etc. Optional and type-specific.
+- **subpath**: extra subpath within a package, relative to the package root.
+  Optional.
+
+
+Components are designed such that they for a hierarchy from the most significant
+on the left left to the least significant components to the right.
+
+
+A `purl` must NOT contain a URL Authority i.e. there is no support for
+`username`, `password`, `host` and `port` components. A `namespace` segment may
+sometimes look like a `host` but its interpretation is specific to a `type`.
+
+
+Some `purl` examples
+~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    bitbucket:birkenfeld/pygments-main@244fd47e07d1014f0aed9c
+
+    deb:curl@7.50.3-1?arch=i386&distro=jessie
+
+    docker:cassandra@sha256:244fd47e07d1004f0aed9c
+    docker:customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io
+
+    gem:jruby-launcher@1.1.2?platform=java
+    gem:ruby-advisory-db-check@0.12.4
+
+    github:package-url/purl-spec@244fd47e07d1004f0aed9c
+
+    go:google.golang.org/genproto#googleapis/api/annotations
```

Comment:
Maybe use gopkg or golang? The term "go" is avoided in a lot of places because it's hard to search for (without excess noise) and this would avoid the conflict you mentioned with the go: prefix already being registered.

Reply:
This makes a lot of sense. So should we use `golang:` or `gopkg:` as a package URLs `type` for Go?

@ashcrow and @jasinner for @victims ...
@R2wenD2 @vbatts and @elad165  for @Grafeas ... which way would you "Go"?

@griesemer @robpike @ken as the designated Golang designers... and @sdboyer as the lead contributor to `go dep` what would be your take and preference? Your input is highly valuable!

For reference, we are specifying here a simple package URLs to conveniently reference a software package in a mostly universal way across tools and techs as a compact URL/URI string. 

The question is: which URL scheme should we use to identity Go "packages"?
e.g. should we use `golang:` or `gopkg:` as a `purl` `type` for Go? Or something else?

And the neater `go://` scheme is already an officially IANA-registered URI/URL scheme (See https://github.com/package-url/purl-spec/pull/1#issuecomment-345274238 for details)